### ISSUE TITLE
More issues with a react-native app

### ISSUE THREAD

**Issue by mahlon-gumbs** [Subito-it/PodBuilder#25]
Hi there.

So I've run into a few issues using this module with react-native. I've managed to get it working when I build with Xcode but not via ci (using fastlane).  I'll walk you through what I've done with a few posts.

### Initial Steps

_Actually, this is the second attempt but I didn't document the error I got with Flipper and another about `use_frameworks!` not being supported present before starting over._
1. Installed PodBuilder
2. Removed `node_modules` and `ios/Pods`
3. Commented out `Flipper` in `ios/Podfile`
4. Ran `pod_builder init`
5. Ran `pod_buillder build_all`

Everything on the cli completed without issues but Xcode complained: `<shortened>/node_modules/react-native/React/CxxUtils/RCTFollyConvert.h:10:10: 'folly/dynamic.h' file not found`

### Original Podfile
```
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '12.0'

target 'eagle' do
  config = use_native_modules!

  use_react_native!(:path => config["reactNativePath"])

  pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'

  target 'eagleTests' do
    inherit! :complete
    # Pods for testing
  end

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable these next few lines.
  # use_flipper!
  # post_install do |installer|
  #   flipper_post_install(installer)
  # end
end

target 'eagle-tvOS' do
  # Pods for eagle-tvOS

  target 'eagle-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end
end
```
### Podfile generated by PodBuilder
```
# Autogenerated by PodBuilder (https://github.com/Subito-it/PodBuilder)
# Any change to this file should be done on PodBuilder/Podfile

platform :ios, '12.0'

target 'eagle' do
  pod 'AzureAuth', :path => 'PodBuilder/Prebuilt/AzureAuth'
  pod 'DoubleConversion', :path => 'PodBuilder/Prebuilt/DoubleConversion'
  pod 'FBReactNativeSpec', :path => 'PodBuilder/Prebuilt/FBReactNativeSpec'
  pod 'Folly', :path => 'PodBuilder/Prebuilt/Folly'
  pod 'GCDWebServer', :path => 'PodBuilder/Prebuilt/GCDWebServer'
  pod 'RCTTypeSafety', :path => 'PodBuilder/Prebuilt/RCTTypeSafety'
  pod 'RNAzureNotificationHub', :path => 'PodBuilder/Prebuilt/RNAzureNotificationHub'
  pod 'RNCAsyncStorage', :path => 'PodBuilder/Prebuilt/RNCAsyncStorage'
  pod 'RNDateTimePicker', :path => 'PodBuilder/Prebuilt/RNDateTimePicker'
  pod 'RNDeviceInfo', :path => 'PodBuilder/Prebuilt/RNDeviceInfo'
  pod 'RNGestureHandler', :path => 'PodBuilder/Prebuilt/RNGestureHandler'
  pod 'RNReactNativeHapticFeedback', :path => 'PodBuilder/Prebuilt/RNReactNativeHapticFeedback'
  pod 'RNReanimated', :path => 'PodBuilder/Prebuilt/RNReanimated'
  pod 'RNScreens', :path => 'PodBuilder/Prebuilt/RNScreens'
  pod 'RNVectorIcons', :path => 'PodBuilder/Prebuilt/RNVectorIcons'
  pod 'RNZipArchive', :path => 'PodBuilder/Prebuilt/RNZipArchive'
  pod 'React-RCTAnimation', :path => 'PodBuilder/Prebuilt/React-RCTAnimation'
  pod 'React-RCTBlob', :path => 'PodBuilder/Prebuilt/React-RCTBlob'
  pod 'React-RCTImage', :path => 'PodBuilder/Prebuilt/React-RCTImage'
  pod 'React-RCTLinking', :path => 'PodBuilder/Prebuilt/React-RCTLinking'
  pod 'React-RCTNetwork', :path => 'PodBuilder/Prebuilt/React-RCTNetwork'
  pod 'React-RCTSettings', :path => 'PodBuilder/Prebuilt/React-RCTSettings'
  pod 'React-RCTText', :path => 'PodBuilder/Prebuilt/React-RCTText'
  pod 'React-RCTVibration', :path => 'PodBuilder/Prebuilt/React-RCTVibration'
  pod 'React-cxxreact', :path => 'PodBuilder/Prebuilt/React-cxxreact'
  pod 'React-jsi', :path => 'PodBuilder/Prebuilt/React-jsi'
  pod 'React-jsiexecutor', :path => 'PodBuilder/Prebuilt/React-jsiexecutor'
  pod 'React-jsinspector', :path => 'PodBuilder/Prebuilt/React-jsinspector'
  pod 'ReactCommon/turbomodule/core', :path => 'PodBuilder/Prebuilt/ReactCommon'
  pod 'ReactNativeART', :path => 'PodBuilder/Prebuilt/ReactNativeART'
  pod 'ReactNativeDarkMode', :path => 'PodBuilder/Prebuilt/ReactNativeDarkMode'
  pod 'SSZipArchive', :path => 'PodBuilder/Prebuilt/SSZipArchive'
  pod 'WindowsAzureMessaging', :path => 'PodBuilder/Prebuilt/WindowsAzureMessaging'
  pod 'Yoga', :path => 'PodBuilder/Prebuilt/Yoga'
  pod 'appcenter-analytics', :path => 'PodBuilder/Prebuilt/appcenter-analytics'
  pod 'appcenter-core', :path => 'PodBuilder/Prebuilt/appcenter-core'
  pod 'appcenter-crashes', :path => 'PodBuilder/Prebuilt/appcenter-crashes'
  pod 'glog', :path => 'PodBuilder/Prebuilt/glog'
  pod 'react-native-appearance', :path => 'PodBuilder/Prebuilt/react-native-appearance'
  pod 'react-native-blur', :path => 'PodBuilder/Prebuilt/react-native-blur'
  pod 'react-native-maps', :path => 'PodBuilder/Prebuilt/react-native-maps'
  pod 'react-native-netinfo', :path => 'PodBuilder/Prebuilt/react-native-netinfo'
  pod 'react-native-webview', :path => 'PodBuilder/Prebuilt/react-native-webview'
  pod 'rn-fetch-blob', :path => 'PodBuilder/Prebuilt/rn-fetch-blob'
  pod 'AppCenter/Analytics', '=4.1.0'
  pod 'AppCenterReactNativeShared', '=4.0.2'
  pod 'FBLazyVector', :path => '../node_modules/react-native/Libraries/FBLazyVector'
  pod 'RCTRequired', :path => '../node_modules/react-native/Libraries/RCTRequired'
  pod 'React', :path => '../node_modules/react-native'
  pod 'React-Core', :path => '../node_modules/react-native'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-callinvoker', :path => '../node_modules/react-native/ReactCommon/callinvoker'
  pod 'RealmJS', :path => '../node_modules/realm'
  pod 'boost-for-react-native', '=1.63.0'
  pod 'react-native-config', :path => '../node_modules/react-native-config'

  target 'eagleTests' do
  end
end


target 'eagle-tvOS' do

  target 'eagle-tvOSTests' do
  end
end


pre_install do |installer|
  Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_duplicate_framework_and_library_names) {}
  require 'pod_builder/podfile/pre_actions_swizzles'
   
end
```

**Comment by mahlon-gumbs** [comment]
### Next Steps

1. Add `Folly` to `skip_pods`
2. Ran `pod_builder build_all`
Again, no errors on the command line.
Xcode now builds and runs in sim with no issues.
However, fastlane no longer builds. It fails with the following error:
```
❌  ld: library not found for -lFolly
```
while running the following command
```
 set -o pipefail && xcodebuild -workspace ./ios/eagle.xcworkspace -scheme eagle -destination 'generic/platform=iOS' -archivePath <shortened>/Developer/Xcode/Archives/2021-02-22/Eagle-1.4.0.45-dev\ 2021-02-22\ 09.52.16.xcarchive PROVISIONING_PROFILE_SPECIFIER=<redacted> clean archive CODE_SIGN_IDENTITY=<redacted> | tee <shortened>/Library/Logs/gym/Eagle-eagle.log | xcpretty
```
### PodBuilder.json
```
{
  "project_name": "eagle",
  "spec_overrides": {
    "Google-Mobile-Ads-SDK": {
      "module_name": "GoogleMobileAds"
    },
    "glog": {
      "pod_target_xcconfig": {
        "DEFINES_MODULE": "NO"
      }
    },
    "DoubleConversion": {
      "pod_target_xcconfig": {
        "DEFINES_MODULE": "NO"
      }
    },
    "Folly": {
      "pod_target_xcconfig": {
        "DEFINES_MODULE": "NO"
      }
    },
    "Flipper-DoubleConversion": {
      "pod_target_xcconfig": {
        "DEFINES_MODULE": "NO"
      }
    },
    "Flipper-Folly": {
      "pod_target_xcconfig": {
        "DEFINES_MODULE": "NO"
      }
    }
  },
  "skip_licenses": [

  ],
  "skip_pods": [
    "Folly",
    "GoogleMaps",
    "React-RCTFabric",
    "React-Core",
    "React-CoreModules"
  ],
  "force_prebuild_pods": [

  ],
  "build_settings": {
    "ENABLE_BITCODE": "NO",
    "GCC_OPTIMIZATION_LEVEL": "s",
    "SWIFT_OPTIMIZATION_LEVEL": "-Osize",
    "SWIFT_COMPILATION_MODE": "wholemodule",
    "CODE_SIGN_IDENTITY": "",
    "CODE_SIGNING_REQUIRED": "NO",
    "CODE_SIGN_ENTITLEMENTS": "",
    "CODE_SIGNING_ALLOWED": "NO"
  },
  "build_settings_overrides": {
    "SBTUITestTunnelClient": {
      "ENABLE_BITCODE": "NO"
    }
  },
  "build_system": "Latest",
  "library_evolution_support": false,
  "license_filename": "Pods-acknowledgements",
  "restore_enabled": true,
  "allow_building_development_pods": false,
  "use_bundler": false,
  "deterministic_build": false,
  "build_using_repo_paths": true,
  "react_native_project": true
}
```
### Podfile (small change in pod order but all the pods are there)
```
# Autogenerated by PodBuilder (https://github.com/Subito-it/PodBuilder)
# Any change to this file should be done on PodBuilder/Podfile

platform :ios, '12.0'

target 'eagle' do
  pod 'AzureAuth', :path => 'PodBuilder/Prebuilt/AzureAuth'
  pod 'DoubleConversion', :path => 'PodBuilder/Prebuilt/DoubleConversion'
  pod 'FBReactNativeSpec', :path => 'PodBuilder/Prebuilt/FBReactNativeSpec'
  pod 'RCTTypeSafety', :path => 'PodBuilder/Prebuilt/RCTTypeSafety'
  pod 'RNAzureNotificationHub', :path => 'PodBuilder/Prebuilt/RNAzureNotificationHub'
  pod 'RNCAsyncStorage', :path => 'PodBuilder/Prebuilt/RNCAsyncStorage'
  pod 'RNDateTimePicker', :path => 'PodBuilder/Prebuilt/RNDateTimePicker'
  pod 'RNDeviceInfo', :path => 'PodBuilder/Prebuilt/RNDeviceInfo'
  pod 'RNGestureHandler', :path => 'PodBuilder/Prebuilt/RNGestureHandler'
  pod 'RNReactNativeHapticFeedback', :path => 'PodBuilder/Prebuilt/RNReactNativeHapticFeedback'
  pod 'RNReanimated', :path => 'PodBuilder/Prebuilt/RNReanimated'
  pod 'RNScreens', :path => 'PodBuilder/Prebuilt/RNScreens'
  pod 'RNVectorIcons', :path => 'PodBuilder/Prebuilt/RNVectorIcons'
  pod 'React-RCTAnimation', :path => 'PodBuilder/Prebuilt/React-RCTAnimation'
  pod 'React-RCTBlob', :path => 'PodBuilder/Prebuilt/React-RCTBlob'
  pod 'React-RCTImage', :path => 'PodBuilder/Prebuilt/React-RCTImage'
  pod 'React-RCTLinking', :path => 'PodBuilder/Prebuilt/React-RCTLinking'
  pod 'React-RCTNetwork', :path => 'PodBuilder/Prebuilt/React-RCTNetwork'
  pod 'React-RCTSettings', :path => 'PodBuilder/Prebuilt/React-RCTSettings'
  pod 'React-RCTText', :path => 'PodBuilder/Prebuilt/React-RCTText'
  pod 'React-RCTVibration', :path => 'PodBuilder/Prebuilt/React-RCTVibration'
  pod 'React-cxxreact', :path => 'PodBuilder/Prebuilt/React-cxxreact'
  pod 'React-jsiexecutor', :path => 'PodBuilder/Prebuilt/React-jsiexecutor'
  pod 'React-jsinspector', :path => 'PodBuilder/Prebuilt/React-jsinspector'
  pod 'ReactCommon/turbomodule/core', :path => 'PodBuilder/Prebuilt/ReactCommon'
  pod 'ReactNativeART', :path => 'PodBuilder/Prebuilt/ReactNativeART'
  pod 'ReactNativeDarkMode', :path => 'PodBuilder/Prebuilt/ReactNativeDarkMode'
  pod 'SSZipArchive', :path => 'PodBuilder/Prebuilt/SSZipArchive'
  pod 'WindowsAzureMessaging', :path => 'PodBuilder/Prebuilt/WindowsAzureMessaging'
  pod 'Yoga', :path => 'PodBuilder/Prebuilt/Yoga'
  pod 'appcenter-analytics', :path => 'PodBuilder/Prebuilt/appcenter-analytics'
  pod 'appcenter-core', :path => 'PodBuilder/Prebuilt/appcenter-core'
  pod 'appcenter-crashes', :path => 'PodBuilder/Prebuilt/appcenter-crashes'
  pod 'glog', :path => 'PodBuilder/Prebuilt/glog'
  pod 'react-native-appearance', :path => 'PodBuilder/Prebuilt/react-native-appearance'
  pod 'react-native-blur', :path => 'PodBuilder/Prebuilt/react-native-blur'
  pod 'react-native-maps', :path => 'PodBuilder/Prebuilt/react-native-maps'
  pod 'react-native-netinfo', :path => 'PodBuilder/Prebuilt/react-native-netinfo'
  pod 'react-native-webview', :path => 'PodBuilder/Prebuilt/react-native-webview'
  pod 'rn-fetch-blob', :path => 'PodBuilder/Prebuilt/rn-fetch-blob'
  pod 'AppCenter/Analytics', '=4.1.0'
  pod 'AppCenterReactNativeShared', '=4.0.2'
  pod 'FBLazyVector', :path => '../node_modules/react-native/Libraries/FBLazyVector'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'
  pod 'GCDWebServer', '=3.5.4'
  pod 'RCTRequired', :path => '../node_modules/react-native/Libraries/RCTRequired'
  pod 'RNZipArchive', :path => '../node_modules/react-native-zip-archive'
  pod 'React', :path => '../node_modules/react-native'
  pod 'React-Core', :path => '../node_modules/react-native'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-callinvoker', :path => '../node_modules/react-native/ReactCommon/callinvoker'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'RealmJS', :path => '../node_modules/realm'
  pod 'boost-for-react-native', '=1.63.0'
  pod 'react-native-config', :path => '../node_modules/react-native-config'

  target 'eagleTests' do
  end
end


target 'eagle-tvOS' do

  target 'eagle-tvOSTests' do
  end
end


pre_install do |installer|
  Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_duplicate_framework_and_library_names) {}
  require 'pod_builder/podfile/pre_actions_swizzles'
   
end
```

**Comment by mahlon-gumbs** [comment]
Actually I take that back. Xcode builds/runs the develop configuration just fine. However, when I try to archive (which is what we're doing in fastlane, it fails with the same error: `ld: library not found for -lFolly`

So both Xcode and fastlane fails due to the missing lib when using the **Release** build configuration.

We also use `Realm` and `react-native-config` which are both automatically skipped due to their script phase.

So I guess my questions are:

1. Should I really skip Folly?
2. What do we do, in general, with skipped pods when we need to build/archive the app since we aren't supposed to run `pod install`?

**Comment by tcamin** [comment]
PodBuilder's support for react-native has to be considered experimental. That being said it did work without too much work on another reactive native project which might help you debug things further. See https://github.com/Subito-it/PodBuilder/issues/10 for more info (you could compare resulting Podfiles).

```
git clone https://github.com/andreialecu/rn63-newproj.git
cd rn63-newproj
yarn install
pod_builder init
pod_builder build_all
```

No changes were required to successfully compile in all build configuration (no need to delete node_modules folder, no need to comment out Flipper or skip Folly).

What version of rn are you using?

Did you try just removing `use_frameworks!` and running `pod_builder build_all`?

It would help to have a sample repo that shows the error, prebuilding rn projects isn't always straightforward but I would be willing to take a quick look.

HTH

**Comment by mahlon-gumbs** [comment]
Hi @tcamin.

Yes, I did try that but I'll try again. It's possible I fat fingered it somehow. I'm the meantime, I've spoken to the client about this and they would like to know if you'd be ok signing an NDA so they we can send you the full project and work with you on figuring this out. Would you be ok with that?

**Comment by tcamin** [comment]
I think it would be much easier if you could come up with a sample project that exhibits the issue, I'm confident that the source code of the project is not necessary to replicate the issue.

**Comment by mahlon-gumbs** [comment]
Hi, sorry for such a long delay. Sent you an invite to a private repo that demonstrates the issue.

**Comment by mahlon-gumbs** [comment]
Hi @tcamin, I saw that you closed this. Did you ever get a chance to look at the repo I sent to you? We still have the problem, unfortunately.

**Comment by tcamin** [comment]
In the spirit of open source I encourage you once again to share a public sample that replicates the issue so that eventually others can contribute to help to solve your issue. I'm not willing to get involved in closed source proprietary software and therefore I did not accept your invitation. Sorry about that

**Comment by mahlon-gumbs** [comment]
Totally understandable. Spoke with the client again (pointed them to your comment). They are in the process of setting up a public repo and will reach back out (today or in the next couple of days).

Would it be better if they opened a new ticket or can this one be reopened?

**Comment by tcamin** [comment]
We can reopen this one.

**Comment by dagraham7** [comment]
@tcamin @mahlon-gumbs - the public repo can be found here: https://github.com/dagraham7/PipelineExample

**Comment by Wxlevan** [comment]
> PodBuilder's support for react-native has to be considered experimental. That being said it did work without too much work on another reactive native project which might help you debug things further. See #10 for more info (you could compare resulting Podfiles).
> 
> ```
> git clone https://github.com/andreialecu/rn63-newproj.git
> cd rn63-newproj
> yarn install
> pod_builder init
> pod_builder build_all
> ```
> 
> No changes were required to successfully compile in all build configuration (no need to delete node_modules folder, no need to comment out Flipper or skip Folly).
> 
> What version of rn are you using?
> 
> Did you try just removing `use_frameworks!` and running `pod_builder build_all`?
> 
> It would help to have a sample repo that shows the error, prebuilding rn projects isn't always straightforward but I would be willing to take a quick look.
> 
> HTH

This is doesn't work for me.
```
Pod::DSLError
/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/cocoapods-core-1.11.2/lib/cocoapods-core/podfile.rb:335:in `rescue in block in from_ruby'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/cocoapods-core-1.11.2/lib/cocoapods-core/podfile.rb:329:in `block in from_ruby'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/cocoapods-core-1.11.2/lib/cocoapods-core/podfile.rb:50:in `instance_eval'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/cocoapods-core-1.11.2/lib/cocoapods-core/podfile.rb:50:in `initialize'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/cocoapods-core-1.11.2/lib/cocoapods-core/podfile.rb:327:in `new'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/cocoapods-core-1.11.2/lib/cocoapods-core/podfile.rb:327:in `from_ruby'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/cocoapods-core-1.11.2/lib/cocoapods-core/podfile.rb:293:in `from_file'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/cocoapods-1.11.2/lib/cocoapods/config.rb:205:in `podfile'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/pod-builder-4.2.0/lib/pod_builder/analyze.rb:24:in `block in installer_at'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/pod-builder-4.2.0/lib/pod_builder/analyze.rb:22:in `chdir'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/pod-builder-4.2.0/lib/pod_builder/analyze.rb:22:in `installer_at'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/pod-builder-4.2.0/lib/pod_builder/command/build.rb:24:in `call'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/pod-builder-4.2.0/lib/pod_builder/command/build_all.rb:11:in `call'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/pod-builder-4.2.0/exe/pod_builder:368:in `block in parse_commandline'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/pod-builder-4.2.0/exe/pod_builder:367:in `each'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/pod-builder-4.2.0/exe/pod_builder:367:in `parse_commandline'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/pod-builder-4.2.0/exe/pod_builder:383:in `<top (required)>'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/bin/pod_builder:23:in `load'
	/Users/evanwang/.rvm/rubies/ruby-3.0.0/bin/pod_builder:23:in `<main>'
	/Users/evanwang/.rvm/gems/ruby-3.0.0/bin/ruby_executable_hooks:22:in `eval'
	/Users/evanwang/.rvm/gems/ruby-3.0.0/bin/ruby_executable_hooks:22:in `<main>'


Command failed!
```
